src/runtime-rs/crates/hypervisor/src/device/device_manager.rs

```diff
@@ -436,6 +456,56 @@ impl DeviceManager {
 
         Err(anyhow!("ID are exhausted"))
     }
+
+    async fn try_update_device(&mut self, update_config: &DeviceConfig) -> Result<()> {
+        let device_id = match update_config {
+            DeviceConfig::ShareFsCfg(config) => {
+                // Try to find the sharefs device. If found, just return.
+                if let Some(target_id) = self.find_device(config.shared_path.clone()).await {
+                    target_id
+                } else {
+                    return Err(anyhow!(
+                        "no matching device was found to do the update operation"
+                    ));
+                }
+            }
+            // TODO for other Device Type
+            _ => {
+                return Err(anyhow!("update device with unsupported device type"));
+            }
+        };
+
+        // get the original device
+        let target_device = self
+            .get_device_info(&device_id)
+            .await
+            .context("get device failed")?;
+
+        // update device with the updated configuration.
+        let updated_device: ArcMutexDevice = match target_device {
+            DeviceType::ShareFs(mut device) => {
+                if let DeviceConfig::ShareFsCfg(config) = update_config {
+                    // update the mount_config.
+                    device.config.mount_config = config.mount_config.clone();
+                }
+                Arc::new(Mutex::new(device.clone()))
+            }
+            _ => return Err(anyhow!("update unsupported device type")),
+        };
+
+        // do handle update
+        let updating_device = updated_device.clone();
```

Comment:
Do you need a clone here for obtaining the lock? if it is required, maybe rename this to `cloned_device` or `updated_device_copy`. The use of a verb in the name is a bit confusing.

Reply:
I don't see either why you'd need to `clone()` the `Arc` here.

Reply:
fixed